nexus/src/db/datastore.rs

```diff
@@ -425,8 +425,13 @@ impl DataStore {
                     .collect())
             })
             .await
-            .map_err(|e| {
-                Error::internal_error(&format!("Transaction error: {}", e))
+            .map_err(|e| match e {
+                TxnError::CustomError(
+                    RegionAllocateError::NotEnoughDatasets(_),
+                ) => Error::unavail("Not enough datasets to allocate disks"),
```

Comment:
It'd be nice to preserve `e`'s error message here so that we have as much detail as possible in the logs.